runelite-client/src/main/java/net/runelite/client/plugins/microbot/tithefarm/TitheFarmingScript.java

```diff
@@ -357,6 +357,24 @@ private void validateInventory() {
 
 // Helper method to validate run energy and patches
         private void validateRunEnergy() {
+            if (Microbot.getClient().getEnergy() < 6000 &&
+                    hasAllEmptyPatches() && state != RECHARING_RUN_ENERGY &&
+                    !Rs2Player.hasStaminaBuffActive()
+            ) {
+                if(Rs2Inventory.hasItem(ItemID.STAMINA_POTION1)) {
```

Comment:
I think this only works properly with 1 stamina potion in the inventory. When you have multiple pots with different fill amount it will call multiple interacts. This because it will meet the if condition of multiple of the below if statements.